pkgs/applications/misc/splatmoji/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, bash
+, fetchFromGitHub
+, fpm
+, gitUpdater
+, jq
+, pandoc
+, shunit2
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "splatmoji";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "cspeterson";
+    repo = "splatmoji";
+    rev = "v${version}";
+    sha256 = "sha256-fsZ8FhLP3vAalRJWUEi/0fe0DlwAz5zZeRZqAuwgv/U=";
+  };
+
+  nativeBuildInputs = [
+    bash
+    fpm
+    jq
+    pandoc
+    shunit2
+  ];
+
+  postPatch = ''
+    patchShebangs ./build.sh
+    patchShebangs ./test/unit_tests
+  '';
+
+  buildPhase = ''
+    ./build.sh ${version} dir
+  '';
```

Comment:
add `runHook preBuild` and `runHook postBuild`. here and for the other phases too.